fms/training/plugins.py

```diff
@@ -240,7 +253,27 @@ def step(
         model_name = (
             model.__class__.__name__.lower() if self.name is None else self.name
         )
-        model_dict = model.state_dict()
+
+        # For FSDP, consolidate checkpointable data to rank0.
+        # TODO: may also want to support distcp checkpoints which should be faster
+        # to save and load but are harder to use for inference.
+        is_fsdp = isinstance(model, FSDP)
+        if is_fsdp and self.group is None:
+            return
```

Comment:
What exactly is this doing?

Reply:
added a comment to explain but basically we only need to checkpoint on the first shard group in hsdp. group is only set for the first shard group.